examples/lighting-app/nrfconnect/main/Rpc.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "Rpc.h"
+#include "AppTask.h"
+
+#include "main/pigweed_lighting.rpc.pb.h"
+#include "pw_hdlc_lite/encoder.h"
+#include "pw_hdlc_lite/rpc_channel.h"
+#include "pw_hdlc_lite/rpc_packets.h"
+#include "pw_hdlc_lite/sys_io_stream.h"
+#include "pw_rpc/echo_service_nanopb.h"
+#include "pw_rpc/server.h"
+#include "pw_sys_io/sys_io.h"
+#include "pw_sys_io_nrfconnect/init.h"
+#include <array>
+#include <span>
+#include <string_view>
+
+namespace chip {
+namespace rpc {
+
+class LightingService final : public generated::LightingService<LightingService>
+{
+public:
+    pw::Status ButtonEvent(ServerContext & ctx, const chip_rpc_Button & request, chip_rpc_Empty & response)
+    {
+        GetAppTask().ButtonEventHandler(request.action << request.idx /* button_state */, 1 << request.idx /* has_changed */);
+        return pw::Status::OK;
+    }
+};
+
+} // namespace rpc
```

Comment:
Why not leave it open and nest the unnamed namespace inside it? There's no requirement that unnamed namespaces be at the root.

Reply:
done